### PR TITLE
Add Mask Scoring RCNN

### PR DESCRIPTION
**Pull request by chongruo**
Add Mask Scoring RCNN

A new folder is created for mask scoring rcnn since the current pipeline is partial hybridized. Will combine it with mask rcnn after implementing two operators in CUDA and making the whole pipeline hybridized. 

will add the trained models.

### CODE REVIEW

**Comment by mli** [comment]
Job PR-956-9 is done. 
Docs are uploaded to http://gluon-vision-staging.s3-website-us-west-2.amazonaws.com/PR-956/9/index.html 
Code coverage of this PR:  vs. Master:

File: gluoncv/model_zoo/mask_score_rcnn/mask_score_rcnn.py [hunk 1]
```diff
@@ -0,0 +1,1071 @@
...
+                        if k >= len(self.classes) or v >= len(old_classes):
+                            warnings.warn("reuse mapping {}/{} -> {}/{} out of range".format(
+                                k, self.classes, v, old_classes))
+                            continue
+                        new_data[k:k + 1] = old_data[v:v + 1]
+                    # set data to new conv layers
+                    new_params.set_data(new_data)
+
+
+class MaskScoreRCNN(nn.Block):
```
Comment (Jerryzcn) [inline gluoncv/model_zoo/mask_score_rcnn/mask_score_rcnn.py]: Can this inherent from MaskRCNN

File: gluoncv/model_zoo/mask_score_rcnn/mask_score_rcnn.py [hunk 2]
```diff
@@ -0,0 +1,1071 @@
...
+        self._rcnn_max_dets = rcnn_max_dets
+        with self.name_scope():
+            self.mask = Mask(self._batch_size, classes, mask_channels, num_fcn_convs=num_fcn_convs,
+                             norm_layer=norm_layer, norm_kwargs=norm_kwargs)
+            self.mask_score = MaskScore(self._batch_size, classes,
+                                        mask_channels, rcnn_max_dets=rcnn_max_dets)
+
+            roi_size = (self._roi_size[0] * target_roi_scale, self._roi_size[1] * target_roi_scale)
+            self._target_roi_size = roi_size
+            self.mask_target = MaskTargetGenerator(
```
Comment (Jerryzcn) [inline gluoncv/model_zoo/mask_score_rcnn/mask_score_rcnn.py]: you can put MaskTargetGenerator in a set and invoke in the training loop instead, and you should be able to hybridize the model

File: scripts/instance/mask_score_rcnn/train_mask_score_rcnn.py [hunk 3]
```diff
@@ -0,0 +1,511 @@
+"""Train Mask RCNN end to end."""
```
Comment (Jerryzcn) [inline scripts/instance/mask_score_rcnn/train_mask_score_rcnn.py]: merge this with mask rcnn

File: scripts/instance/mask_score_rcnn/eval_mask_rcnn.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
+from __future__ import division
```
Comment (Jerryzcn) [inline scripts/instance/mask_score_rcnn/eval_mask_rcnn.py]: merge with mask rcnn

File: scripts/instance/mask_score_rcnn/demo_mask_rcnn.py [hunk 5]
```diff
@@ -0,0 +1,56 @@
+"""Mask RCNN Demo script."""
```
Comment (Jerryzcn) [inline scripts/instance/mask_score_rcnn/demo_mask_rcnn.py]: merge with mask rcnn

**Pull request closed without merging**